### DISCUSSION TITLE
Slice pettern - initialize with props + default - TS

### DISCUSSION THREAD

**Discussion by john093e** [pmndrs/zustand#2500] [Q&A]
Hello everyone,
Can anyone help me understand how to provide Initial state to a slices by using props ? 

For example in the code below I can provide an initial data to the store in order to initialize it with data from my database, how can I then pass the initialData to the slices so it override the Defaultdata of the slice ?

form-store.ts:
```ts
export interface InitialData {
  step1: Step1Schema
  step2: Step2Schema
  step3: Step3Schema
}

export type StoreSlices = Step1Slice &
  Step2Slice &
  Step3Slice &
  StepSlice &
  SubmitFormSlice

export const createPropertyDrawerStore = (initialData?: InitialData) =>
  createStore<StoreSlices>()((...a) => ({
    ...createStep1(...a),
    ...createStep2(...a),
    ...createStep3(...a),
    ...createStepSlice(...a),
    ...createSubmitFormSlice(...a),
  }))
 ```
 
 store/create-step1-slice.ts: 
 ```ts
 export type Step1Schema = z.infer<typeof Step1FormSchema>

export interface Step1Slice {
  step1: Step1Schema
  setStep1: (data: Step1Schema) => void
}

const defaultState: Step1Schema = {
  object: "contacts",
  propertiesGroupId: "",
  name: "",
  internalName: "",
  description: "",
}

const createStep1Slice: StateCreator<Step1Slice> = (set) => ({
  step1: {
    ...defaultState,
  },
  setStep1: (data) => set((state) => ({ step1: { ...state.step1, ...data } })),
})

export default createStep1Slice
```

**Comment by thomas-worcay** [comment]
Hey, did you figure it out @john093e ?
I'm having the same question right now.

**Reply by dbritto-dev** [reply]
@john093e @thomas-worcay to initialize store with initial state you should use: `useStore` + `createStore`, here you go -> https://docs.pmnd.rs/zustand/guides/initialize-state-with-props

> **Note:** if we talk about next.js then we need to additionally use `useContext` + `Context API`
